ntp-udp/src/lib.rs

```diff
@@ -47,18 +49,34 @@ impl UdpSocket {
         peer_addr = debug(self.as_ref().peer_addr()),
         buf_size = buf.len(),
     ))]
-    pub async fn send(&self, buf: &[u8]) -> io::Result<usize> {
+    pub async fn send(&self, buf: &[u8]) -> io::Result<(usize, Option<NtpTimestamp>)> {
+        let send_size = self.send_help(buf).await?;
+
+        // the send timestamp may never come (when the driver does not support send timestamping)
+        // set a very short timeout to prevent hanging forever. We automatically fall back to a
+        // less accurate timestamp when this function returns None
+        let timeout = std::time::Duration::from_millis(10);
+        match tokio::time::timeout(timeout, self.fetch_send_timestamp()).await {
+            Err(_) => Ok((send_size, None)),
+            Ok(send_timestamp) => Ok((send_size, send_timestamp?)),
+        }
```

Comment:
Why not generate the fallback timestamp here? That would save us from returning an option and simplifies callers.